### PR TITLE
Dhis2 2991

### PR DESCRIPTION
**Pull request by huzvak**

### CODE REVIEW

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityInstanceService.java [hunk 1]
```diff
@@ -221,13 +221,29 @@ void validate( TrackedEntityInstanceQueryParams params )
...
+     */
+    boolean trackedEntityInstanceExistsIncludingDeleted( String uid );
+
+    /**
+     //     * Checks for the existence of a TEI by UID
+     //     *
+     //     * @param uid PSI UID to check for
+     //     * @return true/false depending on result
+     //     */
+//    boolean trackedEntityInstanceExists( String uid, boolean includeDeleted );
```
Comment (larshelge) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityInstanceService.java]: Remove commented out code

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/enrollment/AbstractEnrollmentService.java [hunk 2]
```diff
@@ -689,23 +744,33 @@ public ImportSummary updateEnrollmentForNote( Enrollment enrollment )
     @Override
     public ImportSummary deleteEnrollment( String uid )
     {
-        User user = currentUserService.getCurrentUser();
-        
         ProgramInstance programInstance = programInstanceService.getProgramInstance( uid );
+        String descMsg;
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/enrollment/AbstractEnrollmentService.java]: Set to null or empty string to be explicit.

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityInstanceService.java (hunk 1 above)
Comment (huzvak) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityInstanceService.java]: Fixed

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/enrollment/AbstractEnrollmentService.java (hunk 2 above)
Comment (huzvak) [reply dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/enrollment/AbstractEnrollmentService.java]: Fixed

File: dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/AbstractEventService.java [hunk 3]
```diff
@@ -997,29 +963,42 @@ private ImportSummary updateEvent( Event event, User user, boolean singleValue,
...
-
-        if ( organisationUnit == null )
+        if ( !errors.isEmpty() )
         {
-            organisationUnit = programStageInstance.getOrganisationUnit();
+            return new ImportSummary( ImportStatus.ERROR, errors.toString() ).incrementIgnored();
         }
 
-        List<String> errors = trackerAccessManager.canWrite( user, programStageInstance );
+        //TODO: If change of orgUnit shouldn't be supported, the code below can be removed
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/AbstractEventService.java]: Clean up in-line comments here a bit.

Comment (huzvak) [reply dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/AbstractEventService.java]: Removed

**Pull request merged**